contract/views/account_analytic_account_view.xml

```diff
@@ -39,7 +42,7 @@
                             />
                 </div>
                 <group col="4" attrs="{'invisible': [('recurring_invoices','!=',True)]}">
-                    <field name="contract_template_id" colspan="4"/>
+                    <field name="contract_template_id" colspan="4" domain="['|', ('contract_type', '=', contract_type), ('contract_type', '=', False)]"/>
```

Comment:
When you create a new contract template from the purchase contract, the type is a sale contract. Please provide an default_contract_type=contract_type context